common-libs/internal-api-clients/src/services/KeyStorageApiService.ts

```diff
@@ -30,41 +28,43 @@ type GetSignedCredentialRequest = {
 
 type Env = 'dev' | 'staging' | 'prod'
 
-type ConstructorOptions = GenericConstructorOptions & { keyStorageUrl: string }
+type ConstructorOptions = ClientOptions & { keyStorageUrl: string }
 
-type ApiType = BuildApiType<ParseSpec<typeof keyStorageSpec>>
+const client = createClient(keyStorageSpec)
 
 @profile()
-export default class KeyStorageApiService extends GenericApiService<ApiType> {
+export default class KeyStorageApiService {
+  private readonly options
+
   constructor(options: ConstructorOptions) {
-    super(options.keyStorageUrl, options, keyStorageSpec)
+    this.options = createClientOptions(options.keyStorageUrl, options)
   }
 
-  async storeTemplate(params: ApiType['StoreTemplate']['requestBody']) {
-    return this.execute('StoreTemplate', { params })
+  async storeTemplate(params: GetParams<typeof client.StoreTemplate>) {
+    return client.StoreTemplate(this.options, { params })
```

Comment:
Let's move client creation to the constructor and pass "this.options" to its constructor instead of repeating this argument?

```
export default class KeyStorageApiService {
  constructor(options: ConstructorOptions) {
    const options = createClientOptions(options.keyStorageUrl, options)
    this.client = createClient(keyStorageSpec, options)
  }

  async readMyKey({ accessToken }: { accessToken: string }) {
    return this.client.ReadMyKey({ authorization: accessToken })
  }
```

Reply:
(Fixed)